polaris/competition/_base.py

```diff
@@ -0,0 +1,360 @@
+from datetime import datetime
+from itertools import chain
+from typing import Callable, TypeAlias, Union
+from typing_extensions import Self
+from loguru import logger
+from pydantic import (
+    Field,
+    FieldSerializationInfo,
+    SerializerFunctionWrapHandler,
+    computed_field,
+    field_serializer,
+    field_validator,
+    model_validator,
+)
+from polaris.dataset._subset import Subset
+from polaris.evaluate import Metric
+from polaris.evaluate._predictions import CompetitionPredictions
+from polaris.experimental._dataset_v2 import DatasetV2
+from polaris.utils.dict2html import dict2html
+from polaris.utils.errors import InvalidCompetitionError
+from polaris.utils.misc import listit
+from polaris.utils.types import (
+    HubOwner,
+    TargetType,
+    TaskType,
+    SplitType,
+    HttpUrlString,
+    SlugCompatibleStringType,
+    PredictionsType,
+    HubUser,
+)
+
+ColumnsType: TypeAlias = str | list[str]
+
+
+class CompetitionSpecification(DatasetV2):
```

Comment:
**Comment**
Feels weird that `DatasetV2` is experimental, but this is not. Should we move `DatasetV2` to a "regular" package?

Reply:
Yes, I think it's time. 

Would require some documentation, though. Happy to work on that! Especially if you could work on this PR.